src/Components/Web.JS/src/Boot.Server.ts

```diff
@@ -27,32 +27,21 @@ async function boot(userOptions?: Partial<BlazorOptions>): Promise<void> {
   options.reconnectionHandler = options.reconnectionHandler || window['Blazor'].defaultReconnectionHandler;
   logger.log(LogLevel.Information, 'Starting up blazor server-side application.');
 
-  // Initialize statefully prerendered circuits and their components
-  // Note: This will all be removed soon
   const initialConnection = await initializeConnection(options, logger);
-  const circuits = discoverPrerenderedCircuits(document);
-  for (let i = 0; i < circuits.length; i++) {
-    const circuit = circuits[i];
-    for (let j = 0; j < circuit.components.length; j++) {
-      const component = circuit.components[j];
-      component.initialize();
-    }
-  }
-
   const circuit = await startCircuit(initialConnection);
   if (!circuit) {
-    logger.log(LogLevel.Information, 'No preregistered components to render.');
+    throw new Error('Ciruit failed to start.');
   }
 
   const reconnect = async (existingConnection?: signalR.HubConnection): Promise<boolean> => {
```

Comment:
If you're feeling a bit more adventurous here, you should be able to re-use the original connection (call `start()` on it) and then attempt reconnect. We don't really need to new up a new SignalR connection

Reply:
I'm not feeling adventurous. Trying something different sounds like a great idea in 5.0